internal/examples/chatroom/rooms_test.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type Statement struct {
+	text   string
+	author TestUserInfo
+}
+
+type RoomData struct {
+	convo []Statement
+}
+
+type TestUserInfo struct {
+	Name string
+}
+
+func (u TestUserInfo) getUserId() string {
+	return u.Name
+}
+
+func TestRoomsZero(t *testing.T) {
+	rooms := NewRooms[RoomData, TestUserInfo]()
+	assert.NotNil(t, rooms)
```

Comment:
Rewrote with TDD and `-race` flag to make sure no race conditions and I fully understand the code.